### ISSUE TITLE
No output on compile errors

### ISSUE THREAD

**Issue by mhfs** [tildeio/broccoli-typescript-compiler#14]
Hi there,

I just started using `broccoli-typescript-compiler` (0.3.0) and noticed it doesn't detect compilation errors.

Relevant part of my `Brocfile.js`:

``` js
var compileTypeScript = require('broccoli-typescript-compiler');
var scripts = compileTypeScript(tsDir);
```

If I run `tsc` directly:

```
app/boot.ts(2,22): error TS1005: ',' expected.
```

Running broccoli build or serve works like everything is fine.

Is that expected or a known issue?

Thanks!

**Comment by stefanpenner** [comment]
> Is that expected or a known issue?

no

**Comment by stefanpenner** [comment]
adding `reportDiagnostics: true` provides us with `diagnostics`. My preference would be to "fail" the build entirely if a diagnostic was tripped. That way we report nicely to the browser + console, that their was a failure. 

Some points for future work:
- re-construct source location information, to provide helpful error on screen for users (like babel)
- diagnostic may be about more then 1 file, all broccoli's error handling assumes single file failures. This will likely need to be addressed
- this filter is 1 file in 1 file out, so the multi file diagnostics don't make as much sense yet. We do plan to change this going forward though..

**Comment by mhfs** [comment]
@stefanpenner thanks for the swift response. Sorry if it's a silly question but where would I plug the `reportDiagnostics: true`?

**Comment by stefanpenner** [comment]
I'll likely follow up with a fix this weekend. But you would pass it to `ts.transpileModule`

**Comment by mhfs** [comment]
Got it. I'll wait for the fix then. Thanks for all the work on this plugin.

**Comment by mhfs** [comment]
just for the record, I was giving this a try and found this:

https://github.com/Microsoft/TypeScript/issues/4864

> `transpileModule`/`transpile` currently report only syntactic errors. Semantic errors are not reported since if it will yield tons of false positives i.e. we compile standalone file that has imports or uses something that should exists globally (even types from standard library)

**Comment by stefanpenner** [comment]
> transpileModule/transpile currently report only syntactic errors.

thats fine, i believe that is all we want from this tool. The rest should be handled by the editor?

**Comment by mhfs** [comment]
honest answer: I'm new to all this broccoli stuff and have no idea where the line should be. I'm using vim without any typescript tooling.

something else I found while researching is the custom stuff the angular-cli guys put together to glue broccoli and typescript. they're reporting all errors as I'd expected (not implying that's the right behavior).

https://github.com/angular/angular-cli/blob/master/lib/broccoli/broccoli-typescript.js

**Comment by stefanpenner** [comment]
Ya needs more exploration.

I've been using the typescript plugins with vim, when setup it seems to work well

**Comment by philipbjorge** [comment]
It seems a little strange to have syntactic errors reported by your compilation tool and semantic errors reported by your editor.
I don't think I've ever seen tooling that worked that way...

**Comment by stefanpenner** [comment]
@philipbjorge ya it is strange, i honestly haven't thought much about it. I'm just fixing stuff as I hit it, this does deserve more consideration.

**Comment by ef4** [comment]
It does seem bad to not get complete type checking during builds.

I suspect we shouldn't try to do our own file-by-file conversion. The typescript compiler already has its own efficient rebuild watcher, we should delegate to it. It has a global view and can actually typecheck the whole program.

**Comment by krisselden** [comment]
This should be fixed now and @ef4 it is bad because it also has the effect that named re-exports of types are not properly stripped from build output

**Comment by krisselden** [comment]
Which causes downstream tool problems like failing rollup
